plugins/amd-pmc/fu-amd-pmc-device.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2022 Advanced Micro Devices Inc.
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#include "fu-amd-pmc-device.h"
+struct _FuAmdPmcDevice {
+	FuUdevDevice parent_instance;
+};
+
+G_DEFINE_TYPE(FuAmdPmcDevice, fu_amd_pmc_device, FU_TYPE_UDEV_DEVICE)
+
+static gboolean
+fu_amd_pmc_device_probe(FuDevice *device, GError **error)
+{
+	guint64 program;
+	const gchar *version;
+	g_autoptr(GError) error_local = NULL;
+	g_autoptr(FuUdevDevice) udev_parent = NULL;
+	g_autofree gchar *description = NULL;
+
+	version =
+	    fu_udev_device_get_sysfs_attr(FU_UDEV_DEVICE(device), "smu_fw_version", &error_local);
+	if (version == NULL) {
+		if (g_error_matches(error_local, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+			g_set_error_literal(error,
+					    FWUPD_ERROR,
+					    FWUPD_ERROR_NOT_SUPPORTED,
+					    "unsupported kernel version");
+		else
+			g_propagate_error(error, g_steal_pointer(&error_local));
```

Comment:
needs braces

Reply:
i also think it'd be clearer to just return FALSE early and unindent the g_propagate_error

Reply:
Maybe

    if (g_error_matches(error_local, G_IO_ERROR, G_IO_ERROR_NOT_FOUND)) {
        g_set_error_literal()
        return FALSE;
    }
    g_propagate_error()
    return FALSE;

seems more glibby.

Reply:
Got it, thanks!